### ISSUE TITLE
Usability Issue with Determining Unpublished vs Hidden

### ISSUE THREAD

**Issue by matdave** [modxcms/revolution#16465]
### Summary
The unpublished and hide from menu state are reversed on MODX 3 as compared to MODX 2. 

### Step to reproduce
Open a site tree in MODX 2 and then compare it to a matching site tree in MODX 3

### Observed behavior
On 3.x: 

Hidden  = italic
Unpublished = light color

On 2.x

Hidden = lite color
Unpublished = italic

### Expected behavior
The design should be consistent between the two, or possibly improved on 3.x to make it more obvious what is hidden vs unpublished.

**Comment by Ruslan-Aleev** [comment]
In the 3.x branch it is improved, because in the 2.x branch, for some reason, the unpublished resource looks brighter than the one hidden from the menu, which is illogical and strange for UI / UX. See more https://github.com/modxcms/revolution/issues/13969#issuecomment-450426844 and https://github.com/modxcms/revolution/pull/14832

**Comment by rthrash** [comment]
I tend to lean towards the more (but not complete) consistency with the 2.0 behavior. Published being the primary state of content you want visible to the general public seems much more logical to be the same font style—not to mention the gravity of over a decade of established convention. 

Transparently, even I get confused by this too frequently in the new Manager which is otherwise vastly improved, and I should and do know better! 

In theory, I agree with the rationale, but in practice I think we should find a better solution that provides a better UX for both long-time and new users.

**Comment by rthrash** [comment]
The discussion you helpfully pointed to @Ruslan-Aleev included some commentary on removing deleted resources from the tree, simplifying things further, but the trash manager needing more work. It might be interesting to revisit that as it could lead to one less style in the main resource tree, which I think would be a UX win. Simpler is better!

Unpublished and hide from menu could/should have the same visual style, IMO. There will always be exceptions that change what happens (ignore hidden in Wayfinder, e.g., etc.). But, that's for experienced site builders, IMO. "You have to know the rules before you break them."

Accessibility should be kept in mind, so perhaps we need a screen-reader-only addition attached to the icon to indicate state (or after the tree menu name). 

I am interested in the idea of appending a small (red) dot to the icons (maybe lower right) for the exceptions on resources that won't be visible without a Manager login, and leaving normal Resources alone (it also might not work in practice once people experience it). While I have no data to back it up, I suspect most sites should have more Published than Unpublished pages, so less decoration for the UI and probably a better UX and shorter learning curve for new users. The red dot is a fairly established convention for something that is stopped/not ready/dead in many web apps.

**Comment by matdave** [comment]
>  unpublished resource looks brighter than the one hidden from the menu

I think the logic is a matter of perspective. To me, the "lighter color" is attributable to a state of "hidden". My bug report is more that when people upgrade from 2.x to 3.x or manage multiple sites in varying versions, it creates a major disconnect and usability issue.

While some may think swapping the two was better, I am just opening it for discussion. I feel that there could be a better solution that addresses the issue while making it less difficult for people to adopt.

**Comment by smg6511** [comment]
I'd say it makes perfect sense the way the visual cues are made in 2.x and that 3.x should revert to those. I'd say it's more likely that the lighter gray gets interpreted as less opaque as opposed to brighter, and less opaque is a hint for hidden. I'd be hesitant to add more cues such as a dot, as there's already a lot going on with Resource ids being shown and having 4 different font styles.

That said, and with the reversion made, perhaps a helpful hint could also be to add Hide/Unhide to the contextual menus. That way, if the user has forgotten or is unsure of the Resource status, they'd be able to deduce that from the options available in the context menu (e.g., if you see "Publish" and "Unhide" in the menu, you know the Resource is unpublished and hidden from menus).

Maybe?

**Comment by BigBlockStudios** [comment]
The status in the resource browser have been bugging me for a long time.   What I see my users doing (or not doing) 

1. They have no clue - even if told - the nuances between not-published, hidden, hidden AND not-published. They still call me asking why page don't show up (or show up when not supposed to)

2. They never (NEVER) expunge deleted items - they don't know how, despite being told, they still ask how to get rid of them.  I think deleted items should be removed hidden from the resource browser - if possible with an option to "show deleted" 

Tripped over this issue looking for options to colour the icons in the resource browser (for a client).  Recently did a site where get
Resources was used in "landing pages" to collect all the child resources into a single "page builder" kind of deal - I set up the builder template icon to be a pile of blocks and all the section templates to be a single block (obvious - right?) then set wayfinder to only show the builder template in menus - fool proof right? wrong. they still can't figure it out.  i thought It would be great to be able to apply a colour to the icon in the resource browser based on resource status and/or template

The arguments here are

Colour is a better indicator for end users, trying to differentiate between 2 slightly different shades of gray is hard and terrible for AODA

Icons begin to lose their meaning in large resource trees (this is partially on the developer too) particularly when more than 5 or 6 custom icons get used.  The user can't look at a pile of icons and try to guess what each might mean. Colour could go a long way here.

**Comment by BigBlockStudios** [comment]
@rthrash 

> I am interested in the idea of appending a small (red) dot to the icons (maybe lower right) for the exceptions on resources that won't be visible without a Manager login, and leaving normal Resources alone (it also might not work in practice once people experience it). While I have no data to back it up, I suspect most sites should have more Published than Unpublished pages, so less decoration for the UI and probably a better UX and shorter learning curve for new users. The red dot is a fairly established convention for something that is stopped/not ready/dead in many web apps.

Change the garbage can icon to a labelled tab - icons need context, although, granted the garbage can is fairly ubiquitous.  If it were an actual tab that shows a copy of the the resources tab with the deleted items and the trash management page - even better.
even one step better than that - turn it into a badge that indicates number of soft deleted resources.... (within reason 10+ etc) 

(while we are at it - the elements tab needs love - can't tell you the number of times I've looked at a chunk wondering why it's got so much PHP in it - only to realize I am looking at a snippet with the same or similar name.)

**Comment by rthrash** [comment]
@jaygilmore whipped up a plugin that reverts the styling back to something similar to the previous decade+ of convention with some added contrast to be more accessible, following what @matdave described: hidden = light, unpublished = italic. I added the light to the unpublished bit and it looks like this now. CSS variables control the colors/font styles, and it seems a lot more familar to me now:

<img width="289" alt="Image" src="https://github.com/user-attachments/assets/37fbd3d5-0b59-4154-bbab-d4dab5707b1b" />

I could see an argument for making unpublished even more transparent than hidden, but it's a delicate balancing act…

<img width="281" alt="Image" src="https://github.com/user-attachments/assets/411fc6e3-a785-4a43-aab6-1a67e97a80ee" />

Perhaps the hidden icons stay the full dark value to distinguish they’re still published (which solves the "is it published or not red dot thing" I suggested earlier, at least for me).

<img width="289" alt="Image" src="https://github.com/user-attachments/assets/72f498d4-52e9-46e2-920b-26822e077751" />

**Comment by BigBlockStudios** [comment]
Personally for me it's very difficult to differentiate the italicized text, and as I have mentioned users have no idea what it means, frankly even just 'grayed-out' only sometimes indicates to them that 'somethings up'

I think upping the contrast is a very good low UI/UX impact step and gets us away from that soft low-contrast look that is so hard to read when you are 50+

I do think deleted items have no place in the "live" tree - we are all used to that, deleted items 'go away' if you need it back look in the recycle bin. 

I can't think of a universal context that provides a visual cue as to what is published or not - probably the best route is to stick with your convention and up the contrast level some ;) 

The distinction between hidden and not isn't super useful to users either - more of a developer need.

**Comment by jaygilmore** [comment]
Some good suggestions on targets here: https://community.modx.com/t/coloured-icons-in-the-resource-browser/8324/2?u=smashingred 


I'd need help to make my plugin installable and we'd need to adjust it to make it configurable. It could be relativfely easy to set the colors as configs as an array or as indivudual values.

**Comment by modxcommunity** [comment]
This issue has been mentioned on **MODX Community**. There might be relevant details there:

https://community.modx.com/t/coloured-icons-in-the-resource-browser/8324/4

**Comment by BigBlockStudios** [comment]
@modxcommunity  - yes, that was me ;) 

@jaygilmore  - is there a repo for it? not like I have any time at all ... but who knows when I go down those rabbit holes.

**Comment by rthrash** [comment]
No repo yet @BigBlockStudios, but we're discussing a way to get something easy to maintain to make it easy to release a bunch of little utility things like this that we've done. Here's the code for it if you're curious (it really needs a review of the overuse of `!important` for starters…):

```
<?php
/**
 * 
 * osTreeStyles Plugin
 * A Plugin to use the old-school Revo 2.x Tree Styles
 * Also slightly improves node contrast ratios
 * Fires on the MODX system event onManagerPageBeforeRender
 * @author Jay Gilmore (smashingred)
 * @created 2025-02-07
 *
 * For discussions, see https://github.com/modxcms/revolution/issues/16465
 * TL;DR: hidden = light color     unpublished = italic (+ light color?)
 */

/* @var $modx modX */
/* @var $scriptProperties array */

$eventName = $modx->event->name;

switch ($eventName) {
    case 'OnManagerPageBeforeRender':

        $css = '
#modx-leftbar .icon,
.x-tree-node .icon {
    opacity: 1;
}

.x-tree-node {
    color: #333;
    opactity: 1;
    font-style: normal;

    --color-main: #333;

    --opacity-hidemenu: 0.75;
    --font-style-hidemenu: normal;

    --opacity-unpublished: 0.6;
    --font-style-unpublished: italic;

    --color-deleted: #f00;
    --opacity-deleted: 1;
    --opacity-deleted-icon: 0.333;
    --font-style-deleted: var(--font-style-unpublished);    
}
.x-tree-node a span {
    color: inherit !important;
    opacity: inherit !important;
    font-style: inherit !important;
}
.x-tree-node i.icon,
.x-tree-node i.icon-large,
.x-tree-node a span i.icon,
.x-tree-node a span i.icon-large {
    color: inherit;
    opacity: inherit;
    font-style: normal;
    position: relative;
}
.x-tree-node .hidemenu:not(.unpublished) i.icon,
.x-tree-node .hidemenu i.icon-large {
    color: var(--color-main) !important;
    opacity: 1;
}
.x-tree-node .deleted i.icon,
.x-tree-node .deleted i.icon-large {
    color: var(--color-deleted) !important;
    opacity: var(--opacity-deleted-icon) !important;
}

.x-tree-node .hidemenu,
.x-tree-node .hidemenu a span {
    opacity: var(--opacity-hidemenu) !important;
    font-style: inherit !important;
}
.x-tree-node .unpublished,
.x-tree-node .unpublished a span {
    opacity: var(--opacity-unpublished) !important;
    font-style: var(--font-style-unpublished) !important;
}
.x-tree-node .deleted,
.x-tree-node .deleted a span {
    color: var(--color-deleted) !important;
    opacity: var(--opacity-deleted) !important;
    font-style: var(--font-style-deleted) !important;
}
.x-dd-drag-ghost a span,
.x-dd-drag-ghost a {
    color: var(--color-main) !important;
    font-style: inherit !important;
}
';
        $html = '<style>'.preg_replace( '/\s+/', ' ', $css).'</style>';
        $modx->controller->addHtml($html);
        break;
}
```

**Comment by rthrash** [comment]
@Mark-H you were involved in the discussions around this original change, iirc. Any thoughts on how to address this in 3.2?

My gut says upgraders from 2.x to 3.2, would be lost with the tree differences. I am lost without the plugin above due to familiarity with the previous 15-20 years of Evolution/Revolution.

For people that have famiiarlized with 3.x, I suspect a decent, if not majority, would welcome back the tree styles above.

There definitely will be people who will have adopted 3.0/3.1 and like the new styles.

I think the “easiest” way forward would be to update the CSS for 3.2, and release a plugin like the above to revert to the 3.0-3.1.x tree styles and make sure it's clear and obvious on the newsletter, blog, and docs. Adding yet another system setting seems overkill, adds complexity, and would need lexicon translations.

**Comment by Mark-H** [comment]
I have no strong opinions on this - to me this is something people will get used to one way or another and all sides of this discussion has some merit. So leaving a plugin to change it to what people prefer is fine with me.

**Comment by jaygilmore** [comment]
The plugin to allow users to opt into the original 3.x styles is by far the easiest route but I'm exploring the idea of a System Setting and conditionally including the few lines needed for an override of the styles for folks already on 3.x using the inversed Hidden/Unpublished shads.

**Comment by jaygilmore** [comment]
Hey folks, I've submitted a PR for this one to **_revert #14832_** and I'd love to know what folks think about that. See: [Revert #14832 and Improve Tree Resource Contrast to Fix #16465](https://github.com/modxcms/revolution/pull/16759).
